### ISSUE TITLE
A trio TTF bugs

### ISSUE THREAD

**Issue by AxemP** [scp-fs2open/fs2open.github.com#732]
Three reports for the price of one!

1) The gr.Fonts[] access appears to be a 0-based table instead of the normal 1-based table that lua typically does. (see first pic)

2) The specified font height does not appear to be correct. In the linked mod, I have 3 font sizes for the Hind font, and when I compare it to Photoshop's own sizes, the in game ones are noticeably smaller. However the game seems to think they're about the right height when it makes a new line for them (the repeated calls to drawString). (see first pic)

3) When you have that script active and go into the campaign screen, if you have multiple campaigns there seems to be a very large gap between campaigns... (see second pic)

Mod: https://drive.google.com/file/d/0B_q0tkX7VcSaaXNsV1NpcUdzWms/view?usp=sharing
Just go to the main hall to see stuff.

Pics: 
![fontproblems](https://cloud.githubusercontent.com/assets/6625164/17350580/7141b5a0-58ed-11e6-9a95-7e0f32f1d99f.png)
![screen0706](https://cloud.githubusercontent.com/assets/6625164/17350581/77456ac8-58ed-11e6-96b7-091d93105e08.jpg)

**Comment by AxemP** [comment]
I have just been informed that off by one thing was fixed already. So two bugs it is then!!

**Comment by AxemP** [comment]
And just a few other quick observations, I realize now that Photoshop does its font sizes in pts while FS probably does it in pixels... But even with that the font sizes still do not make sense. The size 48 letters are around 22 pixels high, and the space between the bottom of the letters and the top of the row above it is 44 pixels. Likewise the size 36 letters are around 16 pixels high and the entire row height is 32 pixels.

**Comment by asarium** [comment]
1) and 3) have been fixed. 3) belongs an annoying class of bugs that will probably appear multiple times in the future. The code is wrote with changing fonts in mind so if a script changes the font it messes a lot of things up because the font is not reset properly when measuring how tall each line should be. However, these bugs are relatively easy to track down so if you encounter one of these just  create a new issue and it should be fixed pretty fast

Now onto issue 2): I believe this is an asset problem. I tried the Arial font I used for testing and that doesn't cause this issue. I also looked at the fonts on the GNOME font viewer and it looks like the line height of the Hind font is messed up:

![screenshot from 2016-08-03 12-22-43](https://cloud.githubusercontent.com/assets/1013983/17362483/8e492164-5975-11e6-9de7-cd431bd4d5f8.png)

For comparison, this is Arial:
![screenshot from 2016-08-03 12-22-39](https://cloud.githubusercontent.com/assets/1013983/17362501/9fa27988-5975-11e6-9bfb-e4d2a4a67cfb.png)

The Hind font has a much higher line spacing than necessary and is also rendered smaller at the same font size. The test with the GNOME font viewer shows that this is not an issue with the way FSO parses these fonts.
